cognite/neat/workflows/steps/lib/cdf_resources.py

```diff
@@ -0,0 +1,262 @@
+import logging
+import time
+from typing import Tuple
+from cognite.neat.graph.loaders import upload_labels
+from cognite.neat.graph.loaders.core.rdf_to_assets import (
+    NeatMetadataKeys,
+    categorize_assets,
+    rdf2assets,
+    remove_non_existing_labels,
+    unique_asset_labels,
+    upload_assets,
+)
+from cognite.neat.graph.loaders.core.rdf_to_relationships import (
+    categorize_relationships,
+    rdf2relationships,
+    upload_relationships,
+)
+from cognite.neat.graph.loaders.validator import validate_asset_hierarchy
+from cognite.neat.workflows.model import FlowMessage, WorkflowConfigs
+from cognite.neat.workflows.steps.step_model import Step
+from cognite.client.data_classes import AssetFilter
+
+from cognite.client import CogniteClient
+from ..data_contracts import CategorizedAssets, CategorizedRelationships, RulesData, SolutionGraph
+
+# with contextlib.suppress(ValueError):
+#     prom_cdf_resource_stats = Gauge(
+#         "neat_graph_to_asset_hierarchy_wf_cdf_resource_stats",
+#         "CDF resource stats before and after running fast_graph workflow",
+#         ["resource_type", "state"],
+#     )
+# with contextlib.suppress(ValueError):
+#     prom_data_issues_stats = Gauge("neat_graph_to_asset_hierarchy_wf_data_issues", "Data validation issues", ["type"])
+
+
+__all__ = [
+    "CreateCDFLabels",
+    "GenerateCDFAssetsFromGraph",
+    "GenerateCDFRelationshipsFromGraph",
+    "UploadCDFAssets",
+    "UploadCDFRelationships",
+]
+
+
+class CreateCDFLabels(Step):
+    description = "The step creates default NEAT labels in CDF"
+    category = "cdf_resources"
+
+    def run(self, rules: RulesData, cdf_client: CogniteClient) -> None:
+        upload_labels(cdf_client, rules.rules, extra_labels=["non-historic", "historic"])
```

Comment:
Lets make the extra labels config.
```suggestion
class CreateCDFLabels(Step):
    """
    The step creates default NEAT labels in CDF

    Args:
        extra_labels: Extra labels to be created in CDF. Defaults to ["non_historic", "historic"].
    """

    description = "The step creates default NEAT labels in CDF"
    category = "cdf_resources"

    def __init__(self, extra_labels: list[str] = None, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.extra_labels = extra_labels or ["non-historic", "historic"]

    def run(self, rules: RulesData, cdf_client: CogniteClient) -> None:
        upload_labels(cdf_client, rules.rules, extra_labels=self.extra_labels)
```

The `*args`/ `**kwargs` are there for the base class, but I think the base class should not take `init` parameters, and then these would disappear.